src/Base/Core/Interval.hs

```diff
@@ -161,23 +159,23 @@ instance Invertible Interval where
 -- the interval's distance is increased by that integer without
 -- changing its size.
 --
--- TODO: Figure out what to do if dim and aug < 11 and this function
--- receives an integer argument > 11.
---
 -- prop> intervalToDistance i + x == intervalToDistance (i <+> x)
 -- prop> getSize (i <+> _) == getSize i
 infixl 6 <+>
 (<+>) :: Interval -> Int -> Interval
-Interval iQual i <+> x =
-  Interval (iterate modFunc iQual !! abs x) i
+Interval iQual i <+> x = Interval (iterate smartMod iQual !! abs x) i
   where
+    unwrapMaybeQ :: (Maybe Quality -> Maybe Quality) -> Quality -> Quality
+    unwrapMaybeQ f qual = fromMaybe (baseQuality i) (f $ Just qual)
```

Comment:
Hmm. Not sure how I feel about this function... My intuition is that there's a better way to go for what you're going for here, but this may be related to some other choices elsewhere in the code, so I'll hold my thoughts until then.

Reply:
See my recent commits and comments!